src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1488,10 +1501,12 @@ object Parsers {
         mods
 
     private def addModifier(mods: Modifiers): Modifiers = {
-      val flag = flagOfToken(in.token)
+      val tok = in.token
+      val flag = flagOfToken(tok)
+      val mod = atPos(in.offset) { in.nextToken(); modOfToken(tok) }
```

Comment:
shorter:

    val mod = atPos(in.skipToken()) { modOfToken(tok) }

Same for similar places below.